### PR TITLE
fix(core): cap partner-fusion sequence T before scan hang

### PR DESCRIPTION
**Pull request by ss251**
## Summary

Occupancy-FREE complete hang on `origin/main` `ec035f73ad8e1e0545155e94c3e689d4b8d5d82b`.

`PartnerPolicyFusion.decide_sequence` read `decision_ids.shape[0]` and `jax.lax.scan`'d that T with no ceiling. A spy on origin recorded:

```text
decide_sequence(..., T=2048, ...) -> ORIGIN_REACHED_SCAN T=2048
```

The constructor already bounds `max_partners` at 1024 (the largest non-INT32 integer in this L0 module). Sequence T was the remaining unbounded scan axis. This PR exact-gates JAX arrays before reading `ndim`/`shape` and rejects T outside `[1, 1024]` **before** `jax.lax.scan`. Same bound on `feedback_sequence`.

Not leftover-tax persist / 256 MiB / INT32-only. Not `#1874`. Not clocks / forager / IA / FTL / `optimizers.py`. Not `#1989` `learners.py` (different host; `jnp.arange(num_steps)` learning loops). Not A4 `feature_discovery.py`.

## Expected behavior

Exact JAX rank-1 vectors of length `1..1024` are accepted. `T=1025` and the origin hang-class `T=2048` raise `ValueError` with zero `jax.lax.scan` dispatch. Hostile objects with `ndim`/`shape` hooks raise `TypeError` with zero hook dispatch.

## Scope

- `alberta_framework/core/partner_policy_fusion.py`
- `tests/test_partner_policy_fusion_sequence.py`

## Verification

```text
# red on origin/main ec035f73
decide_sequence(..., T=2048) -> jax.lax.scan xs length 2048

# green at this head
.venv/bin/python -m pytest tests/test_partner_policy_fusion_sequence.py tests/test_partner_policy_fusion.py -q -o addopts=""
# 71 passed
.venv/bin/python -m ruff check alberta_framework/core/partner_policy_fusion.py tests/test_partner_policy_fusion_sequence.py
.venv/bin/python -m mypy alberta_framework/core/partner_policy_fusion.py tests/test_partner_policy_fusion_sequence.py
```

- [x] logs: origin `ec035f73` reached `jax.lax.scan` at T=2048; this head rejects 1025 and 2048 before scan; 71 focused tests passed

- [x] domain-artifact: N/A - fusion sequence hang preflight, no IPMNIST shard

- [x] llm-trajectory: N/A - local fail-before-scan proof, no model-trajectory artifact

- AI assistance: yes

- Model(s) used: xai/grok-4.6

- Agent tooling: Grok CLI via cmux

- Skill path: elizaOS/slopdotcash@6000f7847387c3f046d2c4df071a7c0bb8ddc8d6:skills/contribute-to-asi

- Provenance status: self-reported

AI provider/model: xAI / grok-4.6
Client / agent tooling: Grok CLI via cmux
Contribution skill revision: elizaOS/slopdotcash@6000f7847387c3f046d2c4df071a7c0bb8ddc8d6:skills/contribute-to-asi
Attribution status: self-reported
— [grok-cli-ss251]

### CODE REVIEW

**Comment by lalalune** [comment]
Superseded by merged PR #2060, which consolidates this resource boundary with shared scan/work budgets, combined-work checks, and the broader regression panel.
